func.go

```diff
@@ -1072,6 +1073,47 @@ func funcUniqueBy(v, x interface{}) interface{} {
 	return rs
 }
 
+func funcJoin(v, x interface{}) interface{} {
+	vs, ok := v.([]interface{})
+	if !ok {
+		return &iteratorError{v}
+	}
+	if len(vs) == 0 {
+		return ""
+	}
+
+	ss := make([]string, len(vs))
+	for i, e := range vs {
+		switch e := e.(type) {
+		case nil:
+			ss[i] = ""
+		case string:
+			ss[i] = e
+		case bool:
+			if e {
+				ss[i] = "true"
+			} else {
+				ss[i] = "false"
+			}
+		case int, float64, *big.Int:
+			ss[i] = jsonMarshal(e)
+		default:
+			return &unaryTypeError{name: "join", v: e}
+		}
+	}
+
+	if len(ss) == 1 {
+		return ss[0]
+	}
```

Comment:
This is to allow `[1] | join(1)`, but would be nice to move this check earlier somehow